### ISSUE TITLE
Critical error to render

### ISSUE THREAD

**Issue by LeonardoFeriato** [zhad3/zrenderer#36]
Hi, man. How's going?

I would like, please, another help.
I am trying to use this project on AWS EC2 (Win Server 2022), but in the CLI and Server I always get this message:

**D:\MA\SpriteRobeName.lua:145: table index is nil**
![err](https://github.com/zhad3/zrenderer/assets/37941981/a4b1c0fb-9db9-4725-880d-61eece016df7)

I've done the same test on 2 diferrent PC and this issue doesn't happen. It only happens on AWS EC2.
The most interesting thing is that on EC2 there isn't a D: partition, even after installing GCC and Lua 5.4 nothing changes.

Any suggestions?

**Comment by LeonardoFeriato** [comment]
Problaby is the Win Server, lol

**Comment by LeonardoFeriato** [comment]
Problaby is the Win Server, you can closed this one.

**Comment by zhad3** [comment]
Hmm considering the lua error my first guess would be to check whether the lua file is valid.

Also you shouldn't need to install lua separately as it comes included with zrenderer.

**Comment by zhad3** [comment]
The `D:\MA\SpriteRobeName.lua`  is from within the lua file (probably a lub file). Its the directory from the person who created the lua file.

**Comment by LeonardoFeriato** [comment]
I will try an different approach, this issue can bem close. Thank you again, zhad3

**Comment by LeonardoFeriato** [comment]
> Hmm considering the lua error my first guess would be to check whether the lua file is valid.

Well, I don't know if I understand you.
Now I can simulate on my Win 10, I did a new extraction with the last version (06 Feb 2024) of my 'data.grf' of Brazil Ragnarok Online (bRO) client and the same error happens with different file:
![2024-02-14_08h01_33](https://github.com/zhad3/zrenderer/assets/37941981/02914293-8e10-44da-ae6b-731d310ef215)
 
Look the extracted folder using zExtractor:
![2024-02-14_08h03_00](https://github.com/zhad3/zrenderer/assets/37941981/964b484c-e885-4308-9f5e-d8685aa45e19)

The extraction doesn't show error.

**Comment by zhad3** [comment]
I have no idea how that grf is working. I have downloaded the data.grf (which you sent me) and I investigated the shadowtable.lub.
The following entries are missing in the `npcidentity.lub` which are referenced by `shadowtable.lub`:
```
jobtbl.JT_ABR_BATTLE_WARIOR
jobtbl.JT_ABR_DUAL_CANNON
jobtbl.JT_ABR_MOTHER_NET
jobtbl.JT_ABR_INFINITY
jobtbl.JT_MD_HIDDEN_GROUND01
jobtbl.JT_MD_HIDDEN_GROUND02
```
However, these entries DO exist in the kRO `npcidentity.lub`. Does the client load any additional GRF files? Or does the client contain a `npcidentity.lua` file in the `System/`  directory?

**Comment by LeonardoFeriato** [comment]
>  Does the client load any additional GRF files? Or does the client contain a `npcidentity.lua` file in the `System/` directory?
No, there isn't any additional GRF files or a `npcidentity.lua` file in the `System/` directory.

I have no idea what happens. This file was perfectly extracted:
![2024-02-16_16h22_24](https://github.com/zhad3/zrenderer/assets/37941981/67dd390d-ddd3-4984-8bd3-07b8c635c54d)

If I use combined files with the KRO client, could it work?
I mean, keep LuaFiles514 of the kRO and append sprites from the bRO

**Comment by zhad3** [comment]
> I mean, keep LuaFiles514 of the kRO and append sprites from the bRO.

Yes that will work.

But still I wonder how the bRO client even works with these lua errors. Maybe the bRO client it not using the lua files at all?
